### ISSUE TITLE
Only allow cross platform filenames

### ISSUE THREAD

**Issue by nellh** [bids-standard/legacy-validator#460]
Datasets can pass validation with platform specific filenames. The one I ran into is `:` from a Linux filesystem, which makes the filename invalid on Windows.

[Wikipedia has a nice list of reserved characters.](https://en.wikipedia.org/wiki/Filename#Reserved_characters_and_words) I don't think all of those should be excluded, this set might work? `/\?%*:|<>`

**Comment by chrisgorgo** [comment]
Hmm - I don't think there are any possible BIDS compatible filenames that would allow any of those characters (please correct me if I am wrong). In what context does this pose a problem?

**Comment by nellh** [comment]
I had a generated test path in a derivatives folder that included `:` characters and noticed we'd need to handle this case in either OpenNeuro or the validator to avoid platform issues if Linux / macOS users uploaded files like this. Is derivatives meant to have some filename rules that aren't currently validated? Maybe that's a better issue.

**Comment by chrisgorgo** [comment]
Ok I see - in this case I would be happy to merge such check into the validator.

**Comment by jasmainak** [comment]
yeah this issue has been raised previously for MEG. The derivatives/ folder are not currently checked by the validator, although many people would like to have that checked ...

**Comment by effigies** [comment]
This could be applied to ignored files.
